### PR TITLE
DistributedPubSub may gossip with old cluster

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
Fixed https://github.com/akkadotnet/akka.net/issues/2140

### CODE REVIEW

File: src/contrib/cluster/Akka.Cluster.Tools/PublishSubscribe/Internal/TopicMessages.cs [hunk 1]
```diff
@@ -115,31 +115,45 @@ public override int GetHashCode()
...
         public override int GetHashCode()
         {
-            return Versions != null ? Versions.GetHashCode() : 0;
+            unchecked
+            {
+                int hashCode = 13;
+                foreach (var v in Versions.Values)
+                {
+                    hashCode = hashCode * 17 + v.GetHashCode();
+                }
```
Comment (alexvaluyskiy) [inline src/contrib/cluster/Akka.Cluster.Tools/PublishSubscribe/Internal/TopicMessages.cs]: How to get hashcode for a dictionary?

**Comment by alexvaluyskiy** [comment]
Could be racy. Do no merge it today. I would like to run CI a couple of times

**Comment by alexvaluyskiy** [comment]
Run it three times. It works fine

**Comment by Horusiath** [comment]
Is it ready to merge?

**Comment by alexvaluyskiy** [comment]
Yes

**Pull request merged**